src/daft-dsl/src/expr.rs

```diff
@@ -727,17 +727,23 @@ impl Expr {
                 if_false,
                 predicate,
             } => {
-                let if_true_field = if_true.to_field(schema)?;
-                let if_false_field = if_false.to_field(schema)?;
                 let predicate_field = predicate.to_field(schema)?;
                 if predicate_field.dtype != DataType::Boolean {
                     return Err(DaftError::TypeError(format!(
                         "Expected predicate for if_else to be boolean but received {predicate_field}",
                     )));
                 }
-                match try_get_supertype(&if_true_field.dtype, &if_false_field.dtype) {
-                    Ok(supertype) => Ok(Field::new(if_true_field.name, supertype)),
-                    Err(_) => Err(DaftError::TypeError(format!("Expected if_true and if_false arguments for if_else to be castable to the same supertype, but received {if_true_field} and {if_false_field}")))
+                match predicate.as_ref() {
+                    Expr::Literal(lit::LiteralValue::Boolean(true)) => if_true.to_field(schema),
+                    Expr::Literal(lit::LiteralValue::Boolean(false)) => if_false.to_field(schema),
```

Comment:
The tricky bit for me in this PR is the conditional selection of the resultant column's name... Which feels a little ick.

Though I'm not sure what the best behavior is here. It feels a little icky to me though that depending on the inputs, the output expression name can change.

Reply:
Agreed, I'll keep it consistent with the current behavior to use the `if_true` expression name. 

Though I think this may not be the best user experience either, in fact I'm not sure if there is a way to aptly name an if_else expression. We can consider writing in the if_else docs that it's best practice to use if_else in a with_column, or with an alias